src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -122,6 +122,63 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
 
   Blazor._internal.forceCloseConnection = () => connection.stop();
 
+  Blazor._internal.sendJSDataStream = async (data: ArrayBufferView, streamId: string, maximumIncomingBytes: number) => {
+    // Run the rest in the background, without delaying the completion of the call to sendJSDataStream
+    // otherwise we'll deadlock (.NET can't begin reading until this completes, but it won't complete
+    // because nobody's reading the pipe)
+    setTimeout(async () => {
+      const maxMillisecondsBetweenAcks = 500;
+      let numChunksUntilNextAck = 5;
+      let lastAckTime = new Date().valueOf();
+      try {
+        // Chunk Size is the max SignalR message size less 512 bytes assumed overhead.
+        const chunkSize = maximumIncomingBytes - 512;
```

Comment:
This is somewhat tricky/suspicious.

Reply:
When we send a chunk, we send: `('ReceiveJSDataChunk', streamId, nextChunkData, null);`

This is essentially assuming `ReceiveJSDataChunk` text, `streamId` `number/long` and `null` will total less than 512 bytes (aka. overhead). Leaving us with the SignalR max message size (ie. 32 kb - 512 bytes) for the chunk size (`nextChunkData`). Note this does assume there's no additional size overhead to transfer the chunk.

Reply:
I agree there's a risk of fragility here, but unless we can accurately calculate the overhead, I don't see any better alternative than having this sort of fixed assumption.

The good news is that if 512b turns out to be incorrect, it would be broken for everyone, so we'd know about it, right? There's no way the overhead amount could differ from one app to another, is there?

Reply:
> The good news is that if 512b turns out to be incorrect, it would be broken for everyone, so we'd know about it, right? There's no way the overhead amount could differ from one app to another, is there?

I don't foresee it changing between apps either.